src/com/goide/runconfig/application/GoApplicationRunningState.java

```diff
@@ -52,12 +54,13 @@ protected ProcessHandler startProcess() throws ExecutionException {
     }
     try {
       ProcessOutput processOutput = new ProcessOutput();
+      VirtualFile[] sourceRoots = ModuleRootManager.getInstance(myModule).getSourceRoots(false); //Eventually to support multiple module dependencies
```

Comment:
And after a lot of code this is actually "fix", right? Well, I can't accept it unfortunately. Despite of strange inheritance and duplicating code you've made, here are several problems:

1. Lost an ability to configure main file, so it's impossible to use run configuration in the projects that have several main files.
2. Lost an ability to configure working directory. User can configure it but you will replace it anyway.
3. Source roots. Well, it's just wrong :) Despite of the fact that sourceRoots could be empty (especially in Small IDEs) and you'll get IOOBE in this case. Source roots have no any semantic sense in Go support, first source root could point anywhere, any user can mark any directory or even project root as source root and it doesn't affect (and shouldn't) go-support part, especially it shouldn't affect unconfigurable run configurations.